src/JsonSchema/SchemaFactory.php

```diff
@@ -174,10 +188,9 @@ private function buildPropertySchema(Schema $schema, string $definitionName, str
             $propertySchema['externalDocs'] = ['url' => $iri];
         }
 
-        // TODO: 3.0 support multiple types
-        $type = $propertyMetadata->getBuiltinTypes()[0] ?? null;
+        $types = $propertyMetadata->getBuiltinTypes() ?? [];
 
-        if (!isset($propertySchema['default']) && !empty($default = $propertyMetadata->getDefault()) && (null === $type?->getClassName() || !$this->isResourceClass($type->getClassName()))) {
+        if (!isset($propertySchema['default']) && !empty($default = $propertyMetadata->getDefault()) && (!\count($types) || null === ($className = $types[0]->getClassName()) || !$this->isResourceClass($className))) {
```

Comment:
can't this condition be refactored to be called later when we compute `$valueSchema` and therefore reduce the number of call to the `\count($types)`?